src/server/server_impl.go

```diff
@@ -125,7 +125,7 @@ func (server *server) Start() {
 			return
 		}
 		err = http.Serve(server.debugListener.listener, server.debugListener.debugMux)
-		logger.Infof("Failed to start debug server '%+v'", err)
+		logger.Errorf("Failed to start debug server '%+v'", err)
```

Comment:
Shouldn't you check err above before logging this error?

Reply:
good point! Code was already written like that w/o checking err. Might as well fix while I'm at it.

Reply:
This was fixed in https://github.com/envoyproxy/ratelimit/pull/219. Can you merge main and then I think you can revert this part?

Reply:
I rebased those changes just now.  #219 was a different call to `Serve()` so this code still exists.

In #219 I'm not sure about the need to specifically check the type of error is `http.ErrServerClosed` though, why not simply handle all non-nil errors so we anticipate all possible errors?